### DISCUSSION TITLE
Cypress - Not able to use intercept

### DISCUSSION THREAD

**Discussion by FGirardOMT** [cypress-io/cypress#17636] [Questions and Help]
Hi, I am currently trying to use Cypress on a project, and it would be very helpful to use the intercept feature. But even if I try really hard, I'm not able to make it work! I am using the 8.0.0 version of Cypress. I tried to revert to 6.4.0, but didn't work either.

There is my test, I would like to intercept a POST response message:
  `describe('login test', () => {
      it('should successfully log manually', () => {
      cy.intercept({
        method: 'POST',
        url: '${Cypress.env('apiUrl')}/api/Auth/Login', // Api link to login
      }).as('loginPost')
  
      cy.visit('${Cypress.env('baseUrl')}/login');
  
      cy.get('[cy-data="username"]').type('username');
      cy.get('[cy-data="password"]').type('password');
      cy.get('[cy-data="submitButton"]').click();
      cy.wait(200); // this is working
  
      cy.wait('@loginPost').should('include', 'response.body'); // this isn't working
    });
  });`

There is the error:
![2021_08_06_10_11_36_Window](https://user-images.githubusercontent.com/88547171/128526094-5e102bb9-1b41-403d-9672-27b70eda9f54.png)

I tried many way to create the intercept, but nothing works. 
I tried:
`cy.intercept('POST','${Cypress.env('apiUrl')}/api/Auth/Login').as('loginPost')`
`cy.intercept('POST','/api/Auth/Login').as('loginPost')`
and others.. 
I also tried to create the intercept in a before() before my tests, but that didn't work either.

**Comment by portyu9** [comment]
Hi @FGirardOMT 

There are two things to fix here.

First, register the intercept before the action that sends the request—which you're already doing—but make the URL matcher independent of the API host:

```js
cy.intercept('POST', '**/api/Auth/Login').as('loginPost')

cy.visit(`${Cypress.env('baseUrl')}/login`)

cy.get('[cy-data="username"]').type('username')
cy.get('[cy-data="password"]').type('password')
cy.get('[cy-data="submitButton"]').click()

cy.wait('@loginPost')
  .its('response.statusCode')
  .should('eq', 200)

cy.get('@loginPost')
```

If you want to inspect/assert the response body, use the interception object returned by `cy.wait()`:

```js
cy.wait('@loginPost').then(({ request, response }) => {
  expect(response).to.exist
  expect(response.body).to.exist
})
```

or simply:

```js
cy.wait('@loginPost')
  .its('response.body')
  .should('exist')
```

This assertion from the original code is not correct:

```js
cy.wait('@loginPost').should('include', 'response.body')
```

because `response.body` is a nested property of the interception object, not a string/value that the object should "include".

Also, remove:

```js
cy.wait(200)
```

It is unnecessary—`cy.wait('@loginPost')` waits for the matching request and response.

One other detail: if your actual code really uses quotes like this:

```js
'${Cypress.env("apiUrl")}/api/Auth/Login'
```

then `${...}` is being treated literally. Use a template literal:

```js
`${Cypress.env('apiUrl')}/api/Auth/Login`
```

If `cy.wait('@loginPost')` still reports that no request occurred, inspect the actual POST URL in the browser Network panel and match that URL; `**/api/Auth/Login` is usually the simplest option when only the endpoint path matters.
